src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import CharacterCardSmall from '../characterCardSmall/CharacterCardSmall';
+import style from './Modal.module.scss';
+import { getAllCharacters } from '../../services/Api';
+import { ICharacterSmall } from '../../services/types';
+
+interface ModalProps {
+    isModalVisible: boolean
+    setIsModalVisible: (value: boolean) => void
+    inputedName: string
+}
+
+const Modal:React.FC<ModalProps> = ({isModalVisible, setIsModalVisible, inputedName}) => {
+
+    const [characters, setCharacters] = useState<ICharacterSmall[]>([]);
+
+    useEffect(() => {
+        getAllCharacters().then(result => setCharacters(result))
+    }, [characters.length]);
+
+    return (
+        <>
+        {isModalVisible && (
+            <div className={style.modal_container}>
+                <div className={style.modal_window}>
+                    {characters && 
+                        characters.filter(char => char.name === inputedName).map(filteredChar => 
```

Comment:
1) consider the possibility to get it out from layout cause here is the only one character will be shown, no need to enum here
2) better to use .reduce instead of filter & map here cause of performance cases